### PR TITLE
PE-1643 add api support for l2gw and l2gw connections

### PR DESCRIPTION
**Pull request by joshpoth**

### CODE REVIEW

File: core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronLayer2Gateway.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+            this.m = policy;
+        }
+
+        public Layer2GatewayConcreteBuilder() {
+            this.m = new NeutronLayer2Gateway();
+        }
+
+        @Override
+        protected NeutronLayer2Gateway reference() {
+            return m;
```
Comment (Try75) [inline core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronLayer2Gateway.java:215]: You can reference `m` without using `this.`?
I feel like I might have asked this before.

**Review by Try75 — APPROVED** [review]
This all seems reasonable.

File: core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronLayer2Gateway.java (hunk 1 above)
Comment (joshpoth) [reply core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronLayer2Gateway.java:215]: You can as long as there's not a local variable `m` as well.
```
private Foo m;
setM(Foo m){
    m = m; // won't modify the class field
    this.m = m; // works
}
```

Comment (Try75) [reply core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronLayer2Gateway.java:215]: Thanks. I'm not going to promise to not ask the same question again, but I'll try. :-)

**Pull request merged**